hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ExecutionService.java

```diff
@@ -186,6 +195,7 @@ public void run() {
                         continue;
                     }
                     try {
+                        thread.setContextClassLoader(t.jobClassLoader);
```

Comment:
TCCL should be swapped out when Tasklet is dismissed, to prevent hanging on to the reference.